File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isValidBST = function (root) {
    function helper(root, min, max) {

        if (root === null) {
            return true

        }
        return (root.val > min && root.val < max && helper(root.left, min, root.val) &&
            helper(root.right, root.val, max)
        )
    }
    let max = Number.MAX_SAFE_INTEGER;
    let min = Number.MIN_SAFE_INTEGER;
    return helper(root, min, max)


};

File: 0098-validate-binary-search-tree/0098-validate-binary-search-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool helper(TreeNode *root, long min, long max){
        if (root==NULL){
            return true;
        }
        return ((root->val > min && root->val <max) && helper(root->left, min,root->val) &&  helper(root->right, root->val,max));
    } 
    bool isValidBST(TreeNode* root) {
        long min = LLONG_MIN;
        long max = LLONG_MAX;
        return helper(root, min, max);
        
    }
};